src/main/java/com/cloudbees/plugins/credentials/impl/UsernamePasswordCredentialsImpl.java

```diff
@@ -76,6 +83,9 @@ public UsernamePasswordCredentialsImpl(@CheckForNull CredentialsScope scope,
                                            @CheckForNull String username, @CheckForNull String password) {
         super(scope, id, description);
         this.username = Util.fixNull(username);
+        if(FIPS140.useCompliantAlgorithms() && StringUtils.length(password) < 14) {
+            throw new IllegalArgumentException(Messages.passwordTooShortFIPS());
```

Comment:
```suggestion
            throw new hudson.model.Descriptor.FormException(Messages.passwordTooShortFIPS(), "password");
```
?

Reply:
in the constructor? Why not but I find it weird in a constructor to have this exception which looks more related to form (web action) but maybe I'm biased by the class name and it's implementing `HttpResponse`

Reply:
yup, it means the user gets a friendlier error page that tells them what they did that went wrong (ish) [JENKINS-73771](https://issues.jenkins.io/browse/JENKINS-73771)

Reply:
this will need to note the PR as breaking, as such exception is breaking backward compat.

Reply:
> this will need to note the PR as breaking, as such exception is breaking backward compat.

Add a `compatable since` to the `maven-hpi` configuration so it gets called out in the plugin manager then?

Reply:
done with a3924cecba36fe56f4f708a780a21d07e25c7fb3 
should be ok but not really sure about the random number to use version